### DISCUSSION TITLE
[Question] How to get the coordinate of the position of clicking mouse on MatplotlibChart?

### DISCUSSION THREAD

**Discussion by HuaxiaGoodBoy** [flet-dev/flet#4067] [Q&A] [answered]
### Question

In the following example code, the onclick event cannot be triggered

### Code sample

```python
import matplotlib
import matplotlib.pyplot as plt
import flet as ft
from flet.matplotlib_chart import MatplotlibChart

# Use the WebAgg backend
matplotlib.use("WebAgg")

def main(page: ft.Page):
    def onclick(event):
        """Mouse click event handler"""
        if event.xdata is not None and event.ydata is not None:
            print(f'Mouse click coordinates: x={event.xdata}, y={event.ydata}')

    # Create a figure and axis
    fig, ax = plt.subplots()

    # Plot some data so we can see where the mouse clicks
    ax.plot([1, 2, 3, 4], [10, 20, 25, 30], 'r-')

    # Connect the mouse click event
    cid = fig.canvas.mpl_connect('button_press_event', onclick)

    # Add the Matplotlib chart to the page
    chart = MatplotlibChart(fig, expand=True)
    page.add(chart)

# Start the application
ft.app(target=main, view=ft.WEB_BROWSER)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment, marked as answer]
Matplotlib and Plotly charts available in Flet are non-interactive, because it's just an image of your figure. For interactivity, consider trying the [others](https://flet.dev/docs/controls/charts).

**Reply by HuaxiaGoodBoy** [reply]
Thanks, I think so.
But It's not able to achive the effect as the picture below for other charts
<img width="347" alt="image" src="https://github.com/user-attachments/assets/074f1623-6315-408f-a436-7d0bb1def218">
